### ISSUE TITLE
Bed Change Over Increments is Incorrect

### ISSUE THREAD

**Issue by JohnShelley** [NorthArrowResearch/xsviewer-issues#16]
## The Problem

I can tell the bed change over increments is incorrect.  As I change the increment from 20 miles to 10 miles to 5 miles, the max degradation value doesn't change.  The thickness of the bar changes but it doesn't seem to be taking the other values into the average appropriately.

## Reproduction steps

See the screen shots.

![20mi](https://user-images.githubusercontent.com/41058803/44742601-c68f8400-aaef-11e8-9f8a-76bfdfeab051.jpg)

![5mi](https://user-images.githubusercontent.com/41058803/44742614-ce4f2880-aaef-11e8-8d59-b3a3b844b905.jpg)

## Exception message

```text

Paste the exception message you are getting from the app here. It really helps us. 

```

## Anything else?

You can attach links to datasets.

**Comment by philipbaileynar** [comment]
@JohnShelley I have investigated this and the tool is displaying the bed change value at the downstream edge of each bin. Should it be displaying the **average bed change over the distance of the bin** instead?

**Comment by JohnShelley** [comment]
Yes, it should be displaying the average over the bin.  It should be displaying the average over the bin, which can be computed by subtracting the "cumulative bed change" value for the downstream side of the bin from the "cumulative bed change" value on the upstream side of the bin and then dividing by the length of the bin.

**Comment by philipbaileynar** [comment]
@JohnShelley I have reworked this and will put out a release of what I have. The method you described in your last post does not quite match the method laid out in the 2016/11/30 spreadsheet that we were exchanging. In that spreadsheet the bed elevation change in a bin is the upstream bed change (area / top width) minus the downstream bed change (area / top width) divided by the longitudinal bin length. Here's the existing code:

```csharp
double dsBedChange = dsCVOld.BedElevation - dsCVNew.BedElevation;
double usBedChange = usCVOld.BedElevation - usCVNew.BedElevation;

double avgBedChangeOverIncrement = (usBedChange + dsBedChange) / 2d;
double weightedBedChangeOverIncrement = (usBedChange - dsBedChange) / fBinSize;
cumBedChange += weightedBedChangeOverIncrement;
```

Where:
* `ds` is downstream
* `us` is upstream
* `CV` is control volume
* `fBinSize` is the longitudinal length of the bin

**Note** what's shown above is the reworked code that will be part of release 2.0.13. It is not part of any prior releases.
